app/models/profile.rb

```diff
@@ -100,9 +104,10 @@ def short_ref_id
 
   private
 
-  def in_account(account, policy)
+  def in_account(account, policy, os_version)
     Profile.find_by(account: account, ref_id: ref_id,
                     policy: policy,
+                    os_version: os_version,
```

Comment:
No need to find accounts with OS info. We want to find it even if os_version is `nil`.